crates/core/src/types/block.rs

```diff
@@ -117,15 +174,24 @@ pub struct LegacyTransaction {
 
 impl RLPEncode for LegacyTransaction {
     fn encode(&self, buf: &mut dyn bytes::BufMut) {
-        self.nonce.encode(buf);
-        self.gas_price.encode(buf);
-        self.gas.encode(buf);
-        self.to.encode(buf);
-        self.value.encode(buf);
-        self.data.encode(buf);
-        self.v.encode(buf);
-        self.r.encode(buf);
-        self.s.encode(buf);
+        let encoder = structs::Encoder::new(buf)
+            .encode_field(&self.nonce)
+            .encode_field(&self.gas_price)
+            .encode_field(&self.gas);
+
+        // TODO: implement encode for Option?
```

Comment:
probably a good idea

Reply:
Looking at other clients, this seems to be a special case. To avoid confusion between `Encoder::encode_optional_field` vs `Option`'s `RLPEncode::encode` I just added a `TxKind` struct with `RLPEncode`/`Decode` implementations